src/TypeScript/EslintBridgeClient/EslintBridgeClient.cs

```diff
@@ -83,7 +83,12 @@ public async Task<AnalysisResponse> AnalyzeJs(string filePath, CancellationToken
 
             var responseString = await httpWrapper.PostAsync(BuildServerUri("analyze-js"), analysisRequest, cancellationToken);
 
-            return responseString == null ? null : JsonConvert.DeserializeObject<AnalysisResponse>(responseString);
+            if (string.IsNullOrEmpty(responseString))
+            {
+                throw new ArgumentNullException(nameof(responseString));
+            }
+
+            return JsonConvert.DeserializeObject<AnalysisResponse>(responseString);
```

Comment:
Not too sure about this. If the responseString is an empty string, DeserializeObject will return a null AnalysisResponse.
So if I don't throw an exception here, I'd have to check that the response is null in JavascriptAnalyzer and manually call AnalysisFailure on the notifier (instead of having it only in the catch block).

Reply:
If null/empty is an exceptional (i.e. error) response then throwing seems ok.

However, I'm not sure about the type of the exception. It's not really an argument exception so that seems a bit misleading. What gets logged? Ideally we'd want some general like "JavaScript analysis failed".

Reply:
Hm, perhaps just ArgumentException?
In the end it will reach the notifier and output the generic "analysis failed" message, along with this exception.

Reply:
I merged for now with ArgumentException, I will make a change tomorrow if needed.